vllm/v1/spec_decode/eagle.py

```diff
@@ -20,7 +22,10 @@ def __init__(
         self.vllm_config = vllm_config
         self.num_speculative_tokens = (
             vllm_config.speculative_config.num_speculative_tokens)
+        self.max_model_len = vllm_config.model_config.max_model_len
         self.block_size = vllm_config.cache_config.block_size
+
+        # Optimization: pre-compute and cache the arange tensor.
```

Comment:
maybe we could also pass in one used here: https://github.com/vllm-project/vllm/blob/027b204ff15e803190775e04d39973606a3a7021/vllm/v1/worker/gpu_model_runner.py#L240